src/common/BasedBreadCrumb/BasedBreadCrumb.tsx

```diff
@@ -0,0 +1,60 @@
+"use client";
+import React, { useMemo } from "react";
+import Link from "next/link";
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  BreadcrumbList,
+  BreadcrumbSeparator,
+} from "@/components/ui/breadcrumb";
+import { usePathname } from "next/navigation";
+import useMediaQuery from "@/hooks/use-media-query";
+import { Breadcrumbs } from "@/types";
+
+interface Props {
+  breadcrumbs: Breadcrumbs[];
+}
+
+export const BasedBreadCrumb = ({ breadcrumbs }: Props) => {
+  const pathname = usePathname();
+  const isSmallScreen = useMediaQuery("sm");
+
+  const updatedBreadCrumbs = useMemo(() => {
+    const currenLinkIndex = breadcrumbs.findIndex((x) => x.href === pathname);
+    let updatedLinks = breadcrumbs.slice(0, currenLinkIndex + 1);
+    if (isSmallScreen) {
+      updatedLinks = breadcrumbs.slice(-1);
+    }
+    return updatedLinks;
+  }, [pathname]);
```

Comment:
This seems wrong. We are missing dependencies from `breadcrumbs` and `isSmallScreen`. How is this working?